### PR TITLE
Switched to using new auth service back end.

### PR DESCRIPTION
**Pull request by jaynabonne**

### CODE REVIEW

File: app/connectors/TrustAuthConnector.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class TrustAuthResponseBody(redirectUrl: Option[String] = None)
+
+object TrustAuthResponseBody {
+  implicit val format: Format[TrustAuthResponseBody] = Json.format[TrustAuthResponseBody]
+}
+
+trait TrustAuthResponse
```
Comment (adamconder) [inline app/connectors/TrustAuthConnector.scala:36]: Could this be a sealed trait so improve pattern matching exhaustive-ness?

File: app/connectors/TrustAuthConnector.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+trait TrustAuthConnector {
+  def authorised(utr: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[TrustAuthResponse]
+}
+
+class TrustAuthConnectorImpl @Inject()(http: HttpClient, config: FrontendAppConfig)
+  extends TrustAuthConnector {
+
+  val baseUrl: String = config.trustAuthUrl + "/trusts-auth/authorised/"
+
+  override def authorised(utr: String)
```
Comment (adamconder) [inline app/connectors/TrustAuthConnector.scala:52]: Could this be given a better name, authorisedForTrust or authorisedForUtr?

File: app/connectors/TrustAuthConnector.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                         (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[TrustAuthResponse] = {
+    http.GET[TrustAuthResponseBody](baseUrl + utr).map {
+      case TrustAuthResponseBody(Some(redirectUrl)) => TrustAuthDenied(redirectUrl)
```
Comment (adamconder) [inline app/connectors/TrustAuthConnector.scala:55]: Do we still persist the logging in the backend when we've denied someone?

Comment (jaynabonne) [reply app/connectors/TrustAuthConnector.scala:55]: I didn't remove any logging as such. It should all have been repurposed and moved to the back end. If there's something you want to verify, I can look for that specifically.

**Pull request merged**